include/zephyr/devicetree.h

```diff
@@ -2455,6 +2463,7 @@
 
 /**
  * @defgroup devicetree-generic-foreach "For-each" macros
+ * @since 2.3.0
```

Comment:
There are problems in the other direction as well. These APIs did not show up in v2.3.0. DT_FOREACH_NODE() was not part of the original API rewrite.